Sources/Basic/Result.swift

```diff
@@ -57,6 +57,15 @@ public enum Result<Value, ErrorType: Swift.Error> {
         }
     }
 
+    public func flatMap<U>(_ transform: (Value) -> Result<U, ErrorType>) -> Result<U, ErrorType> {
```

Comment:
This method needs doc comments and tests. Can you propose this as a separate change?